src/generator.cpp

```diff
@@ -1195,119 +1151,99 @@ std::string Generator::GeneratorImpl::generatePiecewiseElseCode(const std::strin
 std::string Generator::GeneratorImpl::generateCode(const AnalyserEquationAstPtr &ast) const
 {
     // Generate the code for the given AST.
+    // Note: we don't handle the case of AnalyserEquationAst::Type::BVAR since
+    //       we don't need to generate any code for it.
 
     std::string code;
 
-    switch (ast->type()) {
-    case AnalyserEquationAst::Type::ASSIGNMENT:
+    if (ast->type() == AnalyserEquationAst::Type::ASSIGNMENT) {
```

Comment:
Use a variable instead of calling the `ast->type()` method many times unnecessarily.

Reply:
Well, `AnalyserEquationAst::type()` simply returns `mPimpl->mType`, so I would expect the compiler to optimise that and make it inline, so not sure it's creating a temporary variable just for that.

Reply:
I have had a look at the assembly produced on my mac, as far as I can tell (my assembly knowledge is not super-awesome) the function call is not getting inlined.  I can share the assembly I looked at to come to this conclusion, if you would like to see if you reach the same conclusion.

Reply:
My assembly knowledge is anything goes back to COR, so... it's anything but current. :) I am therefore happy to take your word for it. Kind of disappointing of a modern compiler not to optimise that kind of call. I remember reading, many years ago, that such functions (which are dead simple) would be inlined by the compielr. Anyway, done in cc44eb9f.